airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -116,6 +117,10 @@ def __init__(self,
         self._connection = self._resolve_connection()
         self._is_yarn = 'yarn' in self._connection['master']
 
+        self._track_driver_state = 'spark://' in self._connection['master'] and self._connection['deploy_mode'] == 'cluster'
```

Comment:
sorry to be picky, how about renaming the variable `_track_driver_state` into `_should_track_driver_status`? 
Since there are no types in python, the starting "should" makes the code more obvious by letting everyone know this is a boolean, and `status` is better aligned with spark's vocabulary => this also makes the code easier to read for newcomers.

Reply:
Also, could you please extract this into a method and add unit tests to it to check that we correctly return true and false for plenty of types of `connection` object? 
I think this is important to convince the airflow community that we are not breaking anything here, since the new mechanism we're introduction must not trigger for all the (client mode) cases for which they are using it => such unit -test is a way of demonstrating that